scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroIO.scala

```diff
@@ -147,13 +208,23 @@ object ParquetAvroIO {
     private[avro] val DefaultNumShards = 0
     private[avro] val DefaultSuffix = ".parquet"
     private[avro] val DefaultCompression = CompressionCodecName.SNAPPY
+    private[avro] val DefaultWindowedFilenameFunction
+      : (Int, Int, BoundedWindow, PaneInfo) => String = (_, _, _, _) =>
+      throw new NotImplementedError(
+        "saveAsDynamicParquetAvroFile for windowed SCollections requires a windowed filename function"
+      )
+    private[avro] val DefaultFilenameFunction: (Int, Int) => String = (_, _) =>
+      throw new NotImplementedError("saveAsDynamicParquetAvroFile requires a filename function")
   }
 
   final case class WriteParam private (
     schema: Schema = WriteParam.DefaultSchema,
     numShards: Int = WriteParam.DefaultNumShards,
     suffix: String = WriteParam.DefaultSuffix,
-    compression: CompressionCodecName = WriteParam.DefaultCompression
+    compression: CompressionCodecName = WriteParam.DefaultCompression,
+    windowedFilenameFunction: (Int, Int, BoundedWindow, PaneInfo) => String =
```

Comment:
If these 2 functions are indeed mutually exclusive as discussed below, this should be an `Option[Either[WindowedFilenameFunction, FilenameFunction]]`?

The semantics being: the `Option[T]` must be defined for dynamic writes, absent for regular writes. And for dynamic writes, either one of the 2 functions must be provided.

Reply:
I like this idea - I'll update implementation to use an Option[Either[]] approach.